AppCenterCrashes/AppCenterCrashesTests/MSMockNsUserDefaults.m

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSMockNsUserDefaults.h"
+#import "MSTestFrameworks.h"
+
+@interface MSMockNsUserDefaults ()
+
+@property(nonatomic) NSMutableDictionary<NSString *, NSObject *> *dictionary;
+@property(nonatomic) id mockNSUserDefaults;
+
+@end
+
+@implementation MSMockNsUserDefaults
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _dictionary = [NSMutableDictionary new];
+
+    // Mock MSUserDefaults shared method to return this instance.
+    _mockNSUserDefaults = OCMClassMock([NSUserDefaults class]);
+    OCMStub([_mockNSUserDefaults standardUserDefaults]).andReturn(self);
+  }
+  return self;
+}
+
+- (void)setObject:(id)anObject forKey:(NSString *)aKey {
+
+  // Don't store nil objects.
+  if (!anObject) {
+    return;
```

Comment:
Should we remove the value in this case?

Reply:
I don't think so. It will break many tests and besides, there's a remove functions for that.